src/Logging/Scrubbing/RegexLogScrubbingRule.cs

```diff
@@ -11,25 +11,41 @@ namespace Microsoft.Omex.Extensions.Logging.Scrubbing
 	internal class RegexLogScrubbingRule : ILogScrubbingRule
 	{
 		private readonly Regex m_regexToReplace;
-		private readonly string m_replacementValue;
+		private readonly MatchEvaluator? m_matchEvaluator;
+		private readonly string? m_replacementValue;
 
 		/// <summary>
 		/// Initializes a new instance of the <see cref="RegexLogScrubbingRule"/> class.
 		/// </summary>
 		/// <param name="regexToReplace">The regular expression specifying the strings to replace.</param>
 		/// <param name="replacementValue">The value with which to replace the matching text.</param>
-		public RegexLogScrubbingRule(string regexToReplace, string replacementValue)
+		public RegexLogScrubbingRule(string regexToReplace, string? replacementValue)
```

Comment:
Could we implement this constructor by calling a more generic one, like this?
```Csharp
public RegexLogScrubbingRule(string regexToReplace, string replacementValue)
=> this(regexToReplace, new MatchEvaluator(m => replacementValue)) { }
```
This would allow us to avoid duplicated logic and nullable fields? 
Or do you worried that it would be less performant?

Reply:
As discussed offline, we'll be sticking with current approach as having match evaluator in both scenarios would cause performance degradation in both places.